drivers: led: Pico W LED driver

mcjelcom opened the pull request:
LED driver utilizing the GPIO pins on top of the AIROC Wireless driver.

Add Pico W LED driver to the DTS bindings in the LED section.

soburi commented:
It is true that this is currently only implemented on the rpi pico w,
but looking at the code, it is not specific to the rpi pico w, and other implementations are possible if an LED is connected to the GPIO of the CYW43439.

jlbwoods commented:
> It is true that this is currently only implemented on the rpi pico w, but looking at the code, it is not specific to the rpi pico w, and other implementations are possible if an LED is connected to the GPIO of the CYW43439.

* WL_GPIO0 on the CYW43439 chip is hardwired to a LED for the RPI Pico W.  Other two pins on Pico W seems to be used as "WL_GPIO1: OP controls the on-board SMPS power save pi" and "WL_GPIO2: IP VBUS sense".   The `led_pico_w` driver will simply things by providing the LED light control to the users.
* The proper way to create a GPIO driver for CYW43439 might be to create a MFD driver for CYW43439; the MFD version of CYW43439 driver will have Wi-Fi, Bluetooth, and GPIO support.  By looking at the tickets, I think the Bluetooth part of the driver is still under-construction.  Also, we are not sure if MFD will be used for the driver.

thedjnK reviewed (commented):
still confused as to why you are making a GPIO driver for a device that has GPIOs and creating it as an LED driver, the manufacturer refers to this as a GPIO, not an LED, it is also applicable to any device and shouldn't inherently be tied to one specific soc

simonguinot commented on drivers/led/led_pico_w.c:
```diff
@@ -0,0 +1,77 @@
+#define GPIO_LED_ON  0x1
+#define GPIO_LED_OFF 0x0
+
+int airoc_gpio_set(int gpio_pin, int gpio_state)
```
`static` ^

Commit:
drivers: led: Pico W LED driver

LED driver utilizing the GPIO pins on top of the AIROC Wireless driver.

Add Pico W LED driver to the DTS bindings in the LED section.

Signed-off-by: John Lin <john.lin@beechwoods.com>
